publishing-sdk/src/main/java/com/ably/tracking/publisher/Mapbox.kt

```diff
@@ -190,8 +190,10 @@ internal class DefaultMapbox(
                 stopTripSession()
                 mapboxReplayer?.finish()
                 val tripHistoryString = retrieveHistory()
-                val historyEvents = ReplayHistoryMapper().mapToReplayEvents(tripHistoryString)
-                locationHistoryListener?.invoke(LocationHistoryData(historyEvents.toGeoJsonMessages()))
+                if (tripHistoryString.isNotEmpty()) {
```

Comment:
For future maintainers, perhaps a comment would help:

```suggestion

                // MapBox's mapToReplayEvents method crashes if passed an empty string,
                // so check it's not empty to prevent that.
                // (see: https://github.com/ably/ably-asset-tracking-android/issues/434)
                if (tripHistoryString.isNotEmpty()) {
```

Reply:
https://github.com/ably/ably-asset-tracking-android/commit/66d38c2f3dda06fd428a15505c052903d5f57e39